soh/src/overlays/actors/ovl_En_Fr/z_en_fr.c

```diff
@@ -819,12 +819,22 @@ void EnFr_SetupFrogSong(EnFr* this, PlayState* play) {
     if (this->frogSongTimer != 0) {
         this->frogSongTimer--;
     } else {
-        this->frogSongTimer = 40;
-        this->ocarinaNoteIndex = 0;
-        func_8010BD58(play, OCARINA_ACTION_FROGS);
-        this->ocarinaNote = EnFr_GetNextNoteFrogSong(this->ocarinaNoteIndex);
-        EnFr_CheckOcarinaInputFrogSong(this->ocarinaNote);
-        this->actionFunc = EnFr_ContinueFrogSong;
+        if (CVarGetInteger("gCustomizeFrogsOcarinaGame", 0)) {
```

Comment:
For modifying current flows (as opposed to adding whole new blocks that don't take the place of vanilla code), we want to have an `if...else` where you check for the CVar guarding the change, encapsulate the changed code in that part of the block, and put the entirety of the original code in the `else` block there. So here, it would end up being something like

```c++
        // Enhancement: Customize Frogs Ocarina Game
        if (CVarGetInteger("gCustomizeFrogsOcarinaGame", 0) == 1) {
            this->frogSongTimer = 40 * CVarGetInteger("gFrogsModifyFailTime", 1);
            if (CVarGetInteger("gInstantFrogsGameWin", 0)) {
                            this->actor.textId = 0x40AC;
            EnFr_SetupReward(this, play, false);
        } else { // vanilla code
            this->frogSongTimer = 40;
            this->ocarinaNoteIndex = 0;
            func_8010BD58(play, OCARINA_ACTION_FROGS);
            this->ocarinaNote = EnFr_GetNextNoteFrogSong(this->ocarinaNoteIndex);
            EnFr_CheckOcarinaInputFrogSong(this->ocarinaNote);
            this->actionFunc = EnFr_ContinueFrogSong;
        }
        // Enhancement end
```
It may end up duplicating code in some places, but it will make understanding what's going on clearer for someone coming back to it later (and also for when we figure out how we're going to handle custom code callouts later on, so the codebase is closer to vanilla). We also want to have at least a cursory commentary on what you're changing, like I did with the comments here. I'm not certain how verbose we want to be with it in the end so, for now, so minimal is probably best.

If you could go ahead and make those changes in the rest of your code, and your other PRs where applicable, I'd appreciate it (thankfully, the rest in this PR are pretty much already in that flow, just need the comments). For an example of what to do for wholly new blocks, see my PR to remove stolen tunics from the button equips: #3375 If you have any other questions, feel free to ping me on Discord.

Reply:
Blocked all the vanilla code off + commented. Thanks!